_test/QueryBuilderFrom.test.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace dokuwiki\plugin\struct\test;
+use dokuwiki\plugin\struct\meta\QueryBuilder;
+
+/**
+ * @group plugin_struct
+ * @group plugins
+ */
+class QueryBuilderFrom_struct_test extends StructTest {
+
+    public function test_join() {
+        $qb = new QueryBuilder();
+
+        $qb->addTable('first', 'T1');
+        $qb->addTable('second', 'T2');
+        $qb->addTable('third', 'T3');
+
+        $qb->addLeftJoin('T2', 'fourth', 'T4' , 'T2.foo=T4.foo');
+
+        $expectedSQL = '
+            SELECT FROM first AS T1, second AS T2, LEFT OUTER JOIN fourth AS T4
```

Comment:
this is wrong SQL, there may not be a comma after T2. I just ran into this. Should be fixed in devel branch.